boltons/dictutils.pyi

```diff
@@ -0,0 +1,118 @@
+    def copy(self) -> OrderedMultiDict: ...
+    @classmethod
+    def fromkeys(
+        cls, keys: list[Any], default: Any | None = ...
+    ) -> OrderedMultiDict: ...
+    def update(self, E: dict[Any, Any] | Iterable, **F: dict[str, Any]) -> None: ...
+    def update_extend(
+        self, E: dict[Any, Any] | Iterable, **F: dict[str, Any]
+    ) -> None: ...
+    def __setitem__(self, k: Hashable, v: Any) -> None: ...
+    def __getitem__(self, k: Hashable) -> Any: ...
+    def __delitem__(self, k: Hashable) -> None: ...
+    def __eq__(self, other: OrderedMultiDict) -> bool: ...
+    def __ne__(self, other: OrderedMultiDict) -> bool: ...
+    def pop(self, k: Hashable, default: Any = ...) -> None: ...
+    def popall(self, k: Hashable, default: Any = ...) -> None: ...
+    def poplast(self, k: Hashable = ..., default: Any = ...) -> None: ...
+    def iteritems(self, multi: bool = ...) -> Generator[Any, None, None]: ...
+    def iterkeys(self, multi: bool = ...) -> Generator[Any, None, None]: ...
+    def itervalues(self, multi: bool = ...) -> Generator[Any, None, None]: ...
+    def todict(self, multi: bool = ...) -> dict[Hashable, Any]: ...
+    def sorted(
+        self, key: Any | None = ..., reverse: bool = ...
+    ) -> OrderedMultiDict: ...
+    def sortedvalues(
+        self, key: Any | None = ..., reverse: bool = ...
+    ) -> OrderedMultiDict: ...
+    def inverted(self) -> OrderedMultiDict: ...
+    def counts(self) -> OrderedMultiDict: ...
+    def keys(self, multi: bool = ...) -> list[Hashable]: ...
+    def values(self, multi: bool = ...) -> list[Any]: ...
+    def items(self, multi: bool = ...) -> list[Generator[Any, None, None]]: ...
+    def __iter__(self) -> Generator[Any, None, None]: ...
+    def __reversed__(self) -> Generator[Any, None, None]: ...
+    def viewkeys(self) -> KeysView: ...
+    def viewvalues(self) -> ValuesView: ...
+    def viewitems(self) -> ItemsView: ...
+
+OMD = OrderedMultiDict
+MultiDict = OrderedMultiDict
+
+class FastIterOrderedMultiDict(OrderedMultiDict):
+    def iteritems(self, multi: bool = ...) -> Generator[Any, None, None]: ...
+    def iterkeys(self, multi: bool = ...) -> Generator[Any, None, None]: ...
+    def __reversed__(self) -> Generator[Any, None, None]: ...
+
+class OneToOne(dict):
+    inv: OneToOne
+    def __init__(self, *a, **kw) -> None: ...
+    @classmethod
+    def unique(cls, *a, **kw): ...
+    def __setitem__(self, key: Hashable, val: Hashable) -> None: ...
+    def __delitem__(self, key: Hashable) -> None: ...
+    def clear(self) -> None: ...
+    def copy(self) -> OneToOne: ...
+    def pop(self, key: Hashable, default: Any = ...) -> Hashable: ...
+    def popitem(self) -> tuple[Hashable, Hashable]: ...
+    def setdefault(self, key, default: Any = ...) -> Hashable: ...
+    def update(
+        self, dict_or_iterable: dict | Iterable, **kw: dict[Hashable, Any]
+    ) -> None: ...
+
+class ManyToMany:
+    data: dict[Hashable, Any]
```

Comment:
`Hashable` is very tricky. Mutable types can have immutable subtypes and immutable subtypes can have mutable ones. So, in reallity people prefer `Any` over `Hashable`.